### ISSUE TITLE
No audio in linux

### ISSUE THREAD

**Issue by berarma** [stepmania/stepmania#1972]
I had Stepmania ~5.1-beta2~ 5.1.0a3 working before on Debian 10. Now I've updated the computer hardware (AMD Ryzen 2400G) and the system packages and it's stopped working. I'm not sure about the cause, everything else is working.

I've tried also with the git branch 5.1-new (built by me), the latest alpha of the moondance project and 5.0.12. It's the same with all versoins.

The game starts, sometimes with sound and sometimes without. The audio stops anyway after the splash screen. Everything works except the audio but when I start playing a song it doesn't start, no steps, it's like paused but the game bars color animations work. I can use the Esc key to bring up the menu and quit.

There's no errors in the console output.

Please, any idea to fix it or debug it?

**Comment by coderjo** [comment]
Can you attach a log file? It might help track down the problem.

Also, have you tried changing what sound system stepmania is using?

**Comment by berarma** [comment]
I didn't know there were logs, I was watching the console output. I've attached the contents of the Logs directory.

How can I change the sound system used?

Thanks for the help.

[stepmania-logs.zip](https://github.com/stepmania/stepmania/files/4361763/stepmania-logs.zip)

**Comment by berarma** [comment]
Sorry, I reported that I had version 5.1-beta2 working before and it was 5.1.0a3 instead that I have installed. Everything is still the same. The log that I've uploaded is for that version.

**Comment by coderjo** [comment]
It looks like it is attempting ALSA first, failing that, then OSS, failing that, then JACK, again failing, and finally settling on PulseAudio.

Are you getting audio though other programs that use PulseAudio?

from info.txt:
> ALSA: Advanced Linux Sound Architecture Driver Version k5.4.0-0.bpo.3-amd64.
> ALSA Driver: 0: HD-Audio Generic [Generic], device 3: HDMI 0 [HDMI 0], 1/1 subdevices avail
> ALSA Driver: 0: HD-Audio Generic [Generic], device 7: HDMI 1 [HDMI 1], 1/1 subdevices avail
> ALSA Driver: 1: HD-Audio Generic [Generic_1], device 0: ALC892 Analog [ALC892 Analog], 0/1 subdevices avail
> ALSA: dsnd_pcm_hw_params_set_access: Invalid argument
> Couldn't load driver ALSA-sw: SetHWParams failed
> Couldn't load driver OSS: RageSoundDriver_OSS: Couldn't open /dev/dsp: No such file or directory
> Couldn't load driver JACK: Couldn't connect to JACK server
> Sound driver: Pulse

**Comment by berarma** [comment]
Yes, I use Pulseaudio for the desktop and it works. The new subject isn't exactly the problem. Besides having no audio the songs don't play, neither the audio nor the steps. I don't know if they can be related.

**Comment by coderjo** [comment]
They are related. The game asks the sound driver what the current playback timestamp is to determine where in the chart it should be. If it isn't getting a change in the timestamp, the arrows don't move.

**Comment by berarma** [comment]
I've found these lines in the syslog:

```
Mar 21 00:13:56 zeus pulseaudio[19514]: E: [alsa-sink-ALC892 Analog] alsa-sink.c: ALSA woke us up to write new data to the device, but there was actually nothing to write.
Mar 21 00:13:56 zeus pulseaudio[19514]: E: [alsa-sink-ALC892 Analog] alsa-sink.c: Most likely this is a bug in the ALSA driver 'snd_hda_intel'. Please report this issue to the ALSA developers.
Mar 21 00:13:56 zeus pulseaudio[19514]: E: [alsa-sink-ALC892 Analog] alsa-sink.c: We were woken up with POLLOUT set -- however a subsequent snd_pcm_avail() returned 0 or another value < min_avail.
...
Mar 21 00:13:58 zeus systemd[4502]: pulseaudio.service: Main process exited, code=killed, status=9/KILL
Mar 21 00:13:58 zeus systemd[4502]: pulseaudio.service: Failed with result 'signal'.
Mar 21 00:13:58 zeus systemd[4502]: pulseaudio.service: Service RestartSec=100ms expired, scheduling restart.
Mar 21 00:13:58 zeus systemd[4502]: pulseaudio.service: Scheduled restart job, restart counter is at 5.
Mar 21 00:13:58 zeus systemd[4502]: Stopped Sound Service.
Mar 21 00:13:58 zeus systemd[4502]: Starting Sound Service...
Mar 21 00:13:58 zeus pulseaudio[19627]: W: [pulseaudio] pid.c: Stale PID file, overwriting.
Mar 21 00:13:59 zeus pulseaudio[19627]: W: [pulseaudio] module-loopback.c: Cannot set requested sink latency of 49,93 ms, adjusting to 100,14 ms
Mar 21 00:13:59 zeus pulseaudio[19627]: W: [pulseaudio] module-loopback.c: Cannot set requested source latency of 49,93 ms, adjusting to 100,14 ms
Mar 21 00:13:59 zeus systemd[4502]: Started Sound Service.
Mar 21 00:13:59 zeus bluetoothd[6474]: Endpoint registered: sender=:1.892 path=/MediaEndpoint/A2DPSource
Mar 21 00:13:59 zeus bluetoothd[6474]: Endpoint registered: sender=:1.892 path=/MediaEndpoint/A2DPSink
Mar 21 00:13:59 zeus pulseaudio[19627]: E: [pulseaudio] backend-ofono.c: Failed to register as a handsfree audio agent with ofono: org.freedesktop.DBus.Error.ServiceUnknown: The name org.ofono was not provided by any .service files
Mar 21 00:14:03 zeus gsd-media-keys[11706]: Unable to get default sink
Mar 21 00:14:03 zeus gsd-media-keys[11706]: Unable to get default source
```

It seems Pulseaudio is crashing, maybe because of the ALSA driver error.

I'm using this computer as an HTPC for Kodi and games and this hasn't happened before. That's why I thought it could be a problem with Stepmania. Any idea what could be happening?

Thanks!

**Comment by berarma** [comment]
I fixed it by going back to kernel 4.19.0-6 in Debian. It doesn't work with kernel 5.4.0-0. Probably a regression in the driver in this kernel.

Thanks for the push in the right direction. And sorry for the noise.

Have a great day!
